apps/aecore/src/aec_block_micro_candidate.erl

```diff
@@ -117,7 +117,9 @@ int_create_block(PrevBlockHash, PrevBlock, KeyBlock, Trees, Txs) ->
     Height = aec_blocks:height(KeyBlock),
     Version = aec_hard_forks:protocol_effective_at_height(Height),
 
-    Time = aeu_time:now_in_msecs(),
+    MicroBlockCycle = aec_governance:micro_block_cycle(),
+    LastTime = aec_blocks:time_in_msecs(PrevBlock),
+    Time = max(aeu_time:now_in_msecs(), MicroBlockCycle + LastTime),
```

Comment:
This seems correct only if [previous block was microblock - not if previous block was key block](https://github.com/aeternity/protocol/blame/216819c3629d333ae455c0fb51d13009cfed961d/consensus/consensus.md#L161-L164).

Reply:
Agreed. I will base the decision on the previous block type.